nncf/quantization/algorithms/weight_compression/algorithm.py

```diff
@@ -403,16 +407,24 @@ def apply(
                 backend_entity=self._backend_entity,
             )
 
+        lora_correction_algo = None
+        description = "Applying Weight Compression"
+        if self._lora_correction:
+            lora_correction_params = self._advanced_parameters.lora_correction_params
+            lora_correction_algo = LoraCorrectionAlgorithm(activations, lora_correction_params)
+            description += " with correction of low-rank adapters"
+
         # Sort weight params to start compression with the bigger constants. This lowers peak memory footprint.
         all_weight_params = sorted(all_weight_params, key=lambda wp: wp.num_weights, reverse=True)
 
         # Compress model using weight compression parameters
         transformed_model = self._backend_entity.transform_model(
             model,
             graph,
-            track(all_weight_params, description="Applying Weight Compression"),
+            track(all_weight_params, description=description),
             scales,
             zero_points,
+            lora_correction_algo,
```

Comment:
From a design point of view, you pass backend agnostic algorithm to the backend specific function. I think, we can avoid such behavior if the transformation model loop will be implemented in the backend agnostc code and the algo backend will have only the atomic compression transformation (per weight). Thus you can call lora correction algorithm into the backend agnostic code, for example in the compression algorithm.

Note: I recommend to refactor this part, but this is not a stopper for merge and can be done in a follow-up PR.